test/gst-vaapi/transcode/transcoder.py

```diff
@@ -11,45 +11,199 @@
+    assert self.mode in ["sw", "hw"], "Invalid test case specification as mode type not valid"
 
-  def convert_toyuv(self):
-    self.decodedYUV = get_media()._test_artifact(
-       "{case}_{width}x{height}_{mode}_{trans_type}.yuv".format(**vars(self)))
-    call("gst-launch-1.0 -vf filesrc location={transcoded}"
-      " ! {gstdecoder2}"
-      " ! videoconvert ! video/x-raw,format=I420"
-      " ! checksumsink2 file-checksum=false qos=false"
-      " frame-checksum=false plane-checksum=false dump-output=true"
-      " dump-location={decodedYUV}".format(**vars(self)))
+    # generate platform list based on test runtime parameters
+    iplats, ireq, _ =  self.get_requirements_data("decode", self.codec, self.mode)
+    platforms = set(iplats)
+    requires = [ireq,]
+
+    for output in self.outputs:
+      codec = output["codec"]
+      mode  = output["mode"]
+      assert mode in ["sw", "hw"], "Invalid test case specification as output mode type not valid"
+      oplats, oreq, _ = self.get_requirements_data("encode", codec, mode)
+      platforms &= set(oplats)
+      requires.append(oreq)
+
+    # create matchers based on command-line filters
+    matchers = [Matcher(s) for s in slash.config.root.run.filter_strings]
+
+    # check if user supplied any platform tag on command line
+    pmatch = any(map(lambda p: any([m.matches(p) for m in matchers]), ALL_PLATFORMS))
+
+    # if user supplied a platform tag, check if this test can support it via the
+    # test param-based required platforms
+    if pmatch and not any(map(lambda p: any([m.matches(p) for m in matchers]), platforms)):
+      slash.skip_test("unsupported platform")
+
+    # check required
+    if not all([t for t,m in requires]):
+      slash.skip_test("One or more software requirements not met")
+
+  def gen_input_opts(self):
+    opts = " gst-launch-1.0 -vf filesrc location={source}"
+    return opts.format(**vars(self))
+
+  def gen_output_opts(self):
+    self.goutputs = dict()
+
+    opts = ""
+    _, _, ffdecparms = self.get_requirements_data("decode", self.codec, self.mode)
+    assert ffdecparms is not None, "gen_output_opts decode parameters are empty"
+    ffdecoder, ffdectype = ffdecparms
 
-    #src file to yuv
-    self.referenceFile = get_media()._test_artifact(
-       "{case}_{width}x{height}_{mode}_ref.yuv".format(**vars(self)))
-    call("gst-launch-1.0 -vf filesrc location={source}"
-      " ! {gstdecoder1}"
+    for n, output in enumerate(self.outputs):
+      codec = output["codec"]
+      mode = output["mode"]
+      _, _, ffencparms = self.get_requirements_data("encode", codec, mode)
+      assert ffencparms is not None, "gen_output_opts encode parameters are empty"
+      ffencoder, ext, profile, ffenctype = ffencparms
+
+      for channel in xrange(output.get("channels", 1)):
+        if self.codec == "vc1":
+          opts += " !'video/x-wmv,profile=(string)advanced'"
+          opts += ",width={width},height={height},framerate=14/1"
```

Comment:
These vc1 settings should be moved to the requirements table as part of the ffdecoder value.